src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCommitter.cs

```diff
@@ -60,10 +61,48 @@ public void Commit(CancellationToken cancellationToken)
                 }
             }
 
-            private void RenameSymbolAndApplyChanges(CancellationToken cancellationToken, out bool clearTrackingSession)
+            public async Task<RenameTrackingSolutionSet> RenameSymbolAsync(CancellationToken cancellationToken)
+            {
+                _renameSymbolTask = Task.Factory.SafeStartNewFromAsync(
+                    () => RenameSymbolWorkerAsync(cancellationToken),
+                    cancellationToken,
+                    TaskScheduler.Default);
+
+                return await _renameSymbolTask.ConfigureAwait(false);
```

Comment:
hmm.. if you want to re-use _renameSymbolTask between preview and invoke, I think you shouldn't capture cancellationtoken, since preview and invoke has 2 different cancellationtoken.